### PR TITLE
[RUNTIME] Clean up/refactor code

### PR DESCRIPTION
**Pull request by kevinji**
We don't really need those `r`, `g`, and `a` vars.

### CODE REVIEW

File: runtime/testy/runtime.py [hunk 1]
```diff
@@ -269,14 +264,14 @@ def runtimeTest(testNames):
...
-  if len(testNames) == 0:
-    print("Running all tests")
-    testNames = allTestNames
-  else:
+  if testNames:
     for testName in testNames:
       if testName not in allTestNames:
         print("Error: {} not found.".format(testName))
         return
+  else:
```
Comment (yizow) [inline runtime/testy/runtime.py]: Style thing i picked up over the summer: 

With if/ else chains, you want to structure the code such that the simpler case comes first, and the more complicated case comes second. It's a readibility thing: that way, people can read the simple case and move on to the complicated case while the conditional clause is still fresh in their mind. 

tl;dr change to `if not testNames` and swap the if/else blocks

File: runtime/testy/runtime.py [hunk 2]
```diff
@@ -300,13 +295,14 @@ def runtimeTest(testNames):
       if PROCESS_NAMES.TCP_PROCESS in allProcesses:
         terminate_process(PROCESS_NAMES.TCP_PROCESS)
       sys.stdout = sys.__stdout__
-      print("{}DONE!".format(" "*(50-len(testName))))
-    if not testSuccess(testFileName):
+      print("{}DONE!".format(" " * (50-len(testName))))
+
+    if testSuccess(testFileName):
```
Comment (yizow) [inline runtime/testy/runtime.py:300]: Oh, this is the perfect example of what i mentioned above. good fix.

File: runtime/testy/studentAPI.py [hunk 3]
```diff
@@ -61,17 +62,16 @@ class Gamepad(StudentAPI):
...
+    super().__init__(toManager, fromManager)
 
   def get_value(self, name, gamepad_number=0):
-    gamepad_dict = self._getSMValue('gamepads')[gamepad_number]
-    if name in self.joysticks.keys():
+    gamepad_dict = self._getSMValue("gamepads")[gamepad_number]
+    if name in self.joysticks:
       return gamepad_dict["axes"][self.joysticks[name]]
-    elif name in self.buttons.keys():
+    if name in self.buttons:
```
Comment (yizow) [inline runtime/testy/studentAPI.py]: why remove the `elif`?

File: runtime/testy/studentAPI.py [hunk 4]
```diff
@@ -109,13 +109,13 @@ def run(self, fn, *args, **kwargs):
   def is_running(self, fn):
     if not inspect.isfunction(fn):
       raise StudentAPIValueError("First argument to Robot.run must be a function")
-    elif not inspect.iscoroutinefunction(fn):
+    if not inspect.iscoroutinefunction(fn):
```
Comment (yizow) [inline runtime/testy/studentAPI.py]: same as above. seems like an unneccessary change

File: runtime/testy/studentAPI.py [hunk 5]
```diff
@@ -143,15 +143,12 @@ def _hibikeSubscribeDevice(self, uid, delay, params):
     self.toManager.put([HIBIKE_COMMANDS.SUBSCRIBE, [uid, delay, params]])
 
   def _hibikeGetUID(self, name):
-    try:
-      return self.sensorMappings.get(name)
-    except:
-      raise StudentAPIKeyError()
+    return self.sensorMappings.get(name)
```
Comment (yizow) [inline runtime/testy/studentAPI.py]: we want to return a studentApiError, not the normal KeyError

Comment (yizow) [reply runtime/testy/studentAPI.py]: allows us to extend functionality more easily.

File: runtime/testy/studentAPI.py [hunk 6]
```diff
@@ -61,17 +62,16 @@ class Gamepad(StudentAPI):
     "joystick_right_y" : 3
   }
 
-  def __init__(self):
-    super().__init__(self.toManager, self.fromManager)
+  def __init__(self, toManager, fromManager):
```
Comment (yizow) [inline runtime/testy/studentAPI.py:65]: this seems like almost a functional change.
@brandonxxlee

**Review by yizow — CHANGES_REQUESTED** [review]
I like your readibility fixes and cleanups.

Few comments re: style.

File: runtime/testy/studentAPI.py (hunk 5 above)
Comment (kevinji) [reply runtime/testy/studentAPI.py]: The code as before wouldn't throw an error in the first place, since we're using `.get` instead of a standard dictionary `[]` access.

File: runtime/testy/studentAPI.py (hunk 3 above)
Comment (kevinji) [reply runtime/testy/studentAPI.py]: I guess I personally find it simpler if we have `return` statements since there's no need for the `el` part.

File: runtime/testy/runtime.py (hunk 1 above)
Comment (kevinji) [reply runtime/testy/runtime.py]: Hmm I was more following the principle of "always use positive `if` conditions with `if`-`else` statements, since they're easier to reason about".

File: runtime/testy/studentAPI.py (hunk 6 above)
Comment (brandonxxlee) [reply runtime/testy/studentAPI.py:65]: From a quick glance it seems to still preserve functionality, but I haven't actually tested it.

File: runtime/testy/runtime.py (hunk 1 above)
Comment (yizow) [reply runtime/testy/runtime.py]: not always. 
And in this case, its approx the same either way.

File: runtime/testy/studentAPI.py (hunk 3 above)
Comment (yizow) [reply runtime/testy/studentAPI.py]: functionally, theyre the same.

I would argue that elif is more clear. One does not need to go "oh this if has a return, so only one of these will execute" 

More conciseness is not always more readibility.

File: runtime/testy/studentAPI.py (hunk 5 above)
Comment (yizow) [reply runtime/testy/studentAPI.py]: ah, good point.

**Review by yizow — CHANGES_REQUESTED** [review]
Still would like the `if`/`else` block swapped and the `elif` statements kept

**Comment by kevinji** [comment]
@yizow Changes should be addressed with the latest commit.

**Review by yizow — CHANGES_REQUESTED** [review]
LGTM

**Review by yizow — APPROVED** [review]
